### ISSUE TITLE
googlenet is not scriptable?

### ISSUE THREAD

**Issue by jerryzh168** [pytorch/vision#1936]
## 🐛 Bug

Looks like googlenet is not scriptable

## To Reproduce

```
import torch
import torchvision
m = torch.jit.script(torchvision.models.googlenet(pretrained=True).eval())
```
got:
```
Traceback (most recent call last):
  File "quantize_resnet.py", line 181, in <module>
    main()
  File "quantize_resnet.py", line 138, in main
    m = torch.jit.script(model.float().eval())
  File "/data/users/jerryzh/anaconda3/envs/py3/lib/python3.7/site-packages/torch/jit/__init__.py", line 1267, in script
    return torch.jit._recursive.create_script_module(obj, torch.jit._recursive.infer_methods_to_compile)
  File "/data/users/jerryzh/anaconda3/envs/py3/lib/python3.7/site-packages/torch/jit/_recursive.py", line 305, in create_script_module
    return create_script_module_impl(nn_module, concrete_type, stubs_fn)
  File "/data/users/jerryzh/anaconda3/envs/py3/lib/python3.7/site-packages/torch/jit/_recursive.py", line 352, in create_script_module_impl
    create_methods_from_stubs(concrete_type, stubs)
  File "/data/users/jerryzh/anaconda3/envs/py3/lib/python3.7/site-packages/torch/jit/_recursive.py", line 279, in create_methods_from_stubs
    concrete_type._create_methods(defs, rcbs, defaults)
  File "/data/users/jerryzh/anaconda3/envs/py3/lib/python3.7/site-packages/torch/jit/_recursive.py", line 578, in compile_unbound_method
    create_methods_from_stubs(concrete_type, (stub,))
  File "/data/users/jerryzh/anaconda3/envs/py3/lib/python3.7/site-packages/torch/jit/_recursive.py", line 279, in create_methods_from_stubs
    concrete_type._create_methods(defs, rcbs, defaults)
RuntimeError: 
Module 'GoogLeNet' has no attribute 'aux1' :
  File "/data/users/jerryzh/anaconda3/envs/py3/lib/python3.7/site-packages/torchvision-0.6.0a0+b2e9565-py3.7-linux-x86_64.egg/torchvision/models/googlenet.py", line 156
        aux_defined = self.training and self.aux_logits
        if aux_defined:
            aux1 = self.aux1(x)
                   ~~~~~~~~~ <--- HERE
        else:
            aux1 = None
'GoogLeNet._forward' is being compiled since it was called from 'GoogLeNet.forward'
  File "/data/users/jerryzh/anaconda3/envs/py3/lib/python3.7/site-packages/torchvision-0.6.0a0+b2e9565-py3.7-linux-x86_64.egg/torchvision/models/googlenet.py", line 200
        # type: (Tensor) -> GoogLeNetOutputs
        x = self._transform_input(x)
        x, aux1, aux2 = self._forward(x)
        ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~ <--- HERE
        aux_defined = self.training and self.aux_logits
        if torch.jit.is_scripting():
```
## Expected behavior

googlenet should be scriptable

## Environment

Collecting environment information...
PyTorch version: 1.5.0a0+5c30f1f
Is debug build: No
CUDA used to build PyTorch: None

OS: CentOS Linux 7 (Core)
GCC version: (GCC) 7.3.1 20180303 (Red Hat 7.3.1-5)
CMake version: version 3.15.4

Python version: 3.7
Is CUDA available: No
CUDA runtime version: No CUDA
GPU models and configuration: No CUDA
Nvidia driver version: No CUDA
cuDNN version: No CUDA

Versions of relevant libraries:
[pip] numpy==1.17.2
[pip] torch==1.5.0a0+4098352
[pip] torchvision==0.6.0a0+b2e9565
[conda] mkl                       2019.4                      243
[conda] mkl-include               2019.4                      243
[conda] torch                     1.5.0a0+627bbb8          pypi_0    pypi
[conda] torchvision               0.6.0a0+f922916          pypi_0    pypi

**Comment by jerryzh168** [comment]
cc @eellison

**Comment by fmassa** [comment]
Thanks for the catch!

This seems to be a problem when pre-trained weights are used, because the pre-trained weights do not have `aux` classifiers in https://github.com/pytorch/vision/blob/3b0b201a0939082477817513036ca4491c191347/torchvision/models/googlenet.py#L99-L101

I tried replacing the `del` in https://github.com/pytorch/vision/blob/3b0b201a0939082477817513036ca4491c191347/torchvision/models/googlenet.py#L57
by
```python
model.aux1 = None
model.aux2 = None
```
and also adding conditionals in 
https://github.com/pytorch/vision/blob/3b0b201a0939082477817513036ca4491c191347/torchvision/models/googlenet.py#L155
and https://github.com/pytorch/vision/blob/3b0b201a0939082477817513036ca4491c191347/torchvision/models/googlenet.py#L166
like
```python
if aux_defined and self.aux1 is not None:
```
but still got the following error
```python
RuntimeError:
cannot call a value of type 'None':
  File "/Users/fmassa/github/vision/torchvision/models/googlenet.py", line 160
        aux_defined = self.training and self.aux_logits
        if aux_defined and self.aux1 is not None:
            aux1 = self.aux1(x)
                   ~~~~~~~~~ <--- HERE
        else:
            aux1 = None
'GoogLeNet._forward' is being compiled since it was called from 'GoogLeNet.forward'
  File "/Users/fmassa/github/vision/torchvision/models/googlenet.py", line 204
        # type: (Tensor) -> GoogLeNetOutputs
        x = self._transform_input(x)
        x, aux1, aux2 = self._forward(x)
        ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~ <--- HERE
        aux_defined = self.training and self.aux_logits
        if torch.jit.is_scripting():
```

@eellison do you know how to fix this?

**Comment by eellison** [comment]
Yep I’ll submit a fix

**Comment by fmassa** [comment]
Fixed via #1949
